### PR TITLE
Persist quality setting over section switches and handle missing deri…

### LINKED ISSUE #2820: MEJS Quality Plugin needs work
The MEJS quality plug-in will not display the quality choice modal if a derivative with the default quality does not exist. That is, unless the default is 'auto', in which case it will look for a derivative with the quality 'auto' or find the first available derivative if 'auto' isn't available.

We need to change the plug-in so that it will find the first available derivative when the default it not there, no matter what the default is.

This change should probably be pushed upstream.

### PR DESCRIPTION
**Pull request by bkeese**
…vatives

Fixes #2820

### CODE REVIEW

File: app/assets/javascripts/media_player_wrapper/mejs4_helper_quality.es6 [hunk 1]
```diff
@@ -33,18 +36,21 @@ class MEJSQualityHelper {
...
+  }
+
+  sendQualitySelection(quality) {
     $.ajax({
       type: 'POST',
       url: '/media_objects/set_session_quality',
-      data: { quality: e.target.value },
+      data: { quality: quality },
       dataType: 'json'
     });
```
Comment (adamjarling) [inline app/assets/javascripts/media_player_wrapper/mejs4_helper_quality.es6]: Does this need any type of callback where the UI needs to respond or do something?  Or is the POST request good enough to live on it's own?

File: vendor/assets/javascripts/mediaelement/plugins/quality-avalon.js [hunk 2]
```diff
@@ -92,16 +92,12 @@
...
-                if (value === 'auto') {
-                  var keyExist = t.keyExist(qualityMap, value);
-                  if (keyExist) {
-                    label = value;
-                  } else {
-                    var keyValue = t.getMapIndex(qualityMap, 0);
-                    label = keyValue.key;
-                  }
-                } else {
+                var keyExist = t.keyExist(qualityMap, value);
```
Comment (adamjarling) [inline vendor/assets/javascripts/mediaelement/plugins/quality-avalon.js]: Are we changing the source code of this quality plugin?  If so, we should make a very explicit comment someplace, or override the file on our end so if we do a plugin update, it doesn't mysteriously break.

File: app/assets/javascripts/media_player_wrapper/mejs4_helper_quality.es6 (hunk 1 above)
Comment (bkeese) [reply app/assets/javascripts/media_player_wrapper/mejs4_helper_quality.es6]: No, the post is good enough.

File: vendor/assets/javascripts/mediaelement/plugins/quality-avalon.js (hunk 2 above)
Comment (bkeese) [reply vendor/assets/javascripts/mediaelement/plugins/quality-avalon.js]: From the name, I thought this was our override. Isn't it?

Comment (adamjarling) [reply vendor/assets/javascripts/mediaelement/plugins/quality-avalon.js]: Ah, yea sorry.  I just saw the `vendor/...` at the beginning.  I think I originally modified this for some weird focus behavior fix.

**Pull request merged**